src/all/mangapluscreators/src/eu/kanade/tachiyomi/extension/all/mangapluscreators/MangaPlusCreators.kt

```diff
@@ -16,94 +18,200 @@ import okhttp3.Request
+        return GET(popularUrl, newHeaders)
     }
 
-    override fun popularMangaParse(response: Response): MangasPage {
-        val result = response.asMpcResponse()
+    override fun popularMangaParse(response: Response): MangasPage = parseMangasPageFromElement(
+        response,
+        "div.item-recent",
+    )
 
-        checkNotNull(result.titles) { EMPTY_RESPONSE_ERROR }
+    fun parseMangasPageFromElement(response: Response, selector: String): MangasPage {
+        val result = response.asJsoup()
 
-        val titles = result.titles.titleList.orEmpty().map(MpcTitle::toSManga)
+        val mangas = result.select(selector).map { element ->
+            popularElementToSManga(element)
+        }
 
-        return MangasPage(titles, result.titles.pagination?.hasNextPage ?: false)
+        return MangasPage(mangas, false)
     }
 
+    fun popularElementToSManga(element: org.jsoup.nodes.Element): SManga {
+        val titleThumbnailUrl = element.select(".image-area img").attr("src")
+        val titleContentId = titleThumbnailUrl.toHttpUrl().pathSegments[2]
+        return SManga.create().apply {
+            title = element.select(".title-area .title").text().toString()
+            thumbnail_url = titleThumbnailUrl
+            setUrlWithoutDomain("/titles/$titleContentId")
+        }
+    }
+
+    // LATEST Section
     override fun latestUpdatesRequest(page: Int): Request {
         val newHeaders = headersBuilder()
             .set("Referer", "$baseUrl/titles/recent/?t=episode")
-            .add("X-Requested-With", "XMLHttpRequest")
             .build()
 
-        val apiUrl = "$API_URL/titles/recent/list".toHttpUrl().newBuilder()
+        val apiUrl = "$API_URL/titles/recent/".toHttpUrl().newBuilder()
             .addQueryParameter("page", page.toString())
-            .addQueryParameter("pageSize", POPULAR_PAGE_SIZE)
             .addQueryParameter("l", lang)
-            .addQueryParameter("c", "episode")
-            .addQueryParameter("isWebview", "false")
-            .addQueryParameter("_", System.currentTimeMillis().toString())
+            .addQueryParameter("t", "episode")
             .toString()
 
         return GET(apiUrl, newHeaders)
     }
 
-    override fun latestUpdatesParse(response: Response): MangasPage = popularMangaParse(response)
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val result = json.decodeFromString<MpcResponse>(response.body.string())
 
-    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
-        val refererUrl = "$baseUrl/keywords".toHttpUrl().newBuilder()
-            .addQueryParameter("q", query)
-            .toString()
+        val titles = result.titles.orEmpty().map { title -> title.toSManga() }
 
-        val newHeaders = headersBuilder()
-            .set("Referer", refererUrl)
-            .add("X-Requested-With", "XMLHttpRequest")
-            .build()
+        // TODO: handle last page of latest
```

Comment:
last page will likely have less manga than other pages, so you can compare the count. even better if you can pass a page count or similar to initial request to control number of manga returned by the server